src/workerd/io/compatibility-date.c++

```diff
@@ -196,8 +201,9 @@ void compileCompatibilityFlags(kj::StringPtr compatDate, capnp::List<capnp::Text
       // it redundant, because at a future date it won't be redundant, and someone could want to
       // set the flag early to make sure they don't forget later.
     }
-    if (dateValidation == CompatibilityDateValidation::CODE_VERSION &&
-        enableByFlag && !enableByDate && enableDate == nullptr) {
+    if ((dateValidation == CompatibilityDateValidation::CODE_VERSION &&
+        enableByFlag && !enableByDate && enableDate == nullptr) ||
+        (isExperimental && !allowExperimentalFeatures)) {
       errorReporter.addError(kj::str(
           "The compatibility flag ", enableFlagName, " is experimental and may break or be "
```

Comment:
This error message would be confusing if returned from Cloudflare. Let's check if `dateValidation == CURRENT_DATE_FOR_CLOUDFLARE`, and if so, use a different error, like:

> The compatibility flag <name> is experimental and cannot yet be used in Workers deployed to Cloudflare.

Reply:
done

Reply:
That's not what I meant. You made it so the whole check now only occurs on Cloudflare, and so now `workerd` can use experimental flags even without passing `--experimental`. We still want `workerd` to enforce the `--experimental` flag as well. What I meant is that we should use a different error message depending on the context.